kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/XStreamMarshaller.java

```diff
@@ -103,7 +116,16 @@ public Class realClass(String elementName) {
                 };
             }
         });
+    }
 
+    private MapperWrapper chainMapperWrappers(List<XStreamMarshallerExtension> extensions, MapperWrapper last) {
+        if (extensions.isEmpty()) {
+            return last;
+        } else {
+            XStreamMarshallerExtension head = extensions.get(0);
+            List<XStreamMarshallerExtension> tail = extensions.subList(1, extensions.size());
```

Comment:
wouldn't it be more readable to replace these two lines with extensions.remove(0) so it will directly have the list shorter of that one item? Assuming the extensions initially will be a copy of the original list.